### PR TITLE
feat: add static property `elements` to entity classes

### PR DESCRIPTION
**Pull request by stockbal**
Enables proper access to all entity elements of the underlying `LinkedDefinition`.

e.g.

```cds
entity Books : cuid {
  title : String;
  stock : Integer;
}
```

new property `elements` allows access to all defined elements of the entity and access the underlying `LinkedDefinition` of those elements.

```ts
import { Book } from "#cds-models/my/bookshop";
import cds from "@sap/cds";

if (Book.elements.ID instanceof cds.builtin.classes.UUID) console.log("ID is of type UUID");
if (Book.elements.stock instanceof cds.builtin.classes.Integer) console.log("stock is of type Integer");
```

### CODE REVIEW

File: lib/components/basedefs.js [hunk 1]
```diff
@@ -13,6 +13,10 @@ const timeRegex = '`${number}${number}:${number}${number}:${number}${number}`'
 const baseDefinitions = new SourceFile('_')
 // FIXME: this should be a library someday
 baseDefinitions.addPreamble(`
+import { entity } from '@sap/cds'
+
+export type ElementsOf<T> = {[name in keyof Required<T>]: entity["elements"][string]}
```
Comment (daogrady) [inline lib/components/basedefs.js:18]: this should be the same as
```ts
import { type } from '@sap/cds'

export type ElementsOf<T> = {[name in keyof Required<T>]: type}
```
, or not?

Comment (stockbal) [reply lib/components/basedefs.js:18]: You are right. I will make the change

Comment (stockbal) [reply lib/components/basedefs.js]: You are right. I will make the change

**Review by daogrady — APPROVED** [review]
Looks good, thanks for this addition! :)

**Pull request merged**